src/Lock.php

```diff
@@ -526,6 +526,176 @@ public static function showForItem(CommonDBTM $item)
+                    'is_dynamic'   => 1
+                ]
+            ]);
+            $nb += $iterator->count();
+
+            //count ipaddress
+            //see Lock.php:412
+            foreach ($iterator as $networNameID) {
+                $iterator = $DB->request([
+                    'SELECT' => 'id',
+                    'FROM'   => IPAddress::getTable(),
+                    'WHERE'  => [
+                        'itemtype'     => 'NetworkName',
+                        'items_id'     => $networNameID,
+                        'mainitems_id' => $item->fields['id'],
+                        'mainitemtype' => $item->getType(),
+                        'is_deleted'   => 1,
+                        'is_dynamic'   => 1
+                    ]
+                ]);
+                $nb += $iterator->count();
+            }
+        }
+
+        //Software versions
+        $iterator = $DB->request([
+            'SELECT'    => [
+                'isv.id AS id',
+                'sv.name AS version',
+                's.name AS software'
+            ],
+            'FROM'      => Item_SoftwareVersion::getTable() . " AS isv",
+            'LEFT JOIN' => [
+                'glpi_softwareversions AS sv' => [
+                    'FKEY' => [
+                        'isv' => 'softwareversions_id',
+                        'sv'  => 'id'
+                    ]
+                ],
+                'glpi_softwares AS s'         => [
+                    'FKEY' => [
+                        'sv'  => 'softwares_id',
+                        's'   => 'id'
+                    ]
+                ]
+            ],
+            'WHERE'     => [
+                'isv.is_deleted'  => 1,
+                'isv.is_dynamic'  => 1,
+                'isv.items_id'     => $item->fields['id'],
+                'isv.itemtype'     => $item->getType(),
+            ]
+        ]);
+        $nb += $iterator->count();
+
+        //Software licenses
+        $iterator = $DB->request([
+            'SELECT'    => [
+                'isl.id AS id',
+                'sl.name AS version',
+                's.name AS software'
+            ],
+            'FROM'      => Item_SoftwareLicense::getTable() . " AS isl",
+            'LEFT JOIN' => [
+                'glpi_softwarelicenses AS sl' => [
+                    'FKEY' => [
+                        'isl' => 'softwarelicenses_id',
+                        'sl'  => 'id'
+                    ]
+                ],
+                'glpi_softwares AS s'         => [
+                    'FKEY' => [
+                        'sl'  => 'softwares_id',
+                        's'   => 'id'
+                    ]
+                ]
+            ],
+            'WHERE'     => [
+                'isl.is_deleted'  => 1,
+                'isl.is_dynamic'  => 1,
+                'isl.items_id'    => $item->fields['id'],
+                'isl.itemtype'    => $item->getType(),
+            ]
+        ]);
```

Comment:
Same remark here, and result is not even used.

Reply:
done !